### PR TITLE
feat(web): Add RSS news feed

### PR DESCRIPTION
**Pull request by Ingimarsson**
# Add RSS news feed

## What

Add an RSS feed under `/rss.xml` that takes in query parameters `tag` and `lang`.

## Why

We currently need this for some organizations, to configure automatic mailing lists in Mailchimp for their news.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/web/pages/api/rss/index.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      <link>https://island.is${
+        linkResolver('news', [newsItem.slug]).href
+      }</link>
+      <description>${newsItem.intro}</description>
+      <pubDate>${new Date(newsItem.date).toUTCString()}</pubDate>
+    </item>`,
+      )
+      .join('')}
+  </channel>
+</rss>`
```
Comment (kristofer-lp8f) [inline apps/web/pages/api/rss/index.ts:49]: Can you break down this nested templating? I would also like to see to see a templated base URL for development.

**Review by kristofer-lp8f — CHANGES_REQUESTED** [review]
I like the change. Nice i18n.

I'm not familiar with RSS internals, so these suggestions/change requests might not apply.

File: apps/web/pages/api/rss/index.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const locale = isLocale(req.query?.lang) ? req.query.lang : defaultLanguage
+
+  const apolloClient = initApollo({}, locale)
+
+  const news = await apolloClient.query<GetNewsQuery, QueryGetNewsArgs>({
+    query: GET_NEWS_QUERY,
+    variables: {
+      input: {
+        lang: locale as ContentLanguage,
+        size: 25,
```
[the remark this answers is not in the document]
Comment (Ingimarsson) [reply apps/web/pages/api/rss/index.ts:25]: I have not seen that with RSS feeds on other sites, maybe we can skip it for now :)

File: apps/web/pages/api/rss/index.ts [hunk 3]
```diff
@@ -27,26 +28,34 @@ export default async function handler(req, res) {
     },
   })
 
+  const baseUrl =
+    process.env.NODE_ENV === 'production'
+      ? 'https://island.is'
+      : 'https://beta.dev01.devland.is'
```
[the remark this answers is not in the document]
Comment (kristofer-lp8f) [reply apps/web/pages/api/rss/index.ts:34]: This should allow any host/proxy:
```suggestion
  const host: string = req.headers.host
  const protocol = `http${host.startsWith('localhost') ? 's' : ''}://`
  const baseUrl = `${protocol}${host}`
```

**Review by kristofer-lp8f — APPROVED** [review]
Good adjustments and PR.

Minor comments/possible improvements, but non-blocking. Removed auto-merge and approving so you can address these comments at your own discretion.

File: apps/web/pages/api/rss/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ingimarsson) [reply apps/web/pages/api/rss/index.ts]: Nice, thank you

File: apps/web/pages/api/rss/index.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      <pubDate>${date}</pubDate>
+    </item>`
+  }
+
+  const feed = `<?xml version="1.0" encoding="UTF-8" ?>
+    <rss version="2.0">
+      <channel>
+        <title>Ísland.is</title>
+        <link>${baseUrl}</link>
+        <description>Ísland.is</description>
```
[the remark this answers is not in the document]
Comment (Ingimarsson) [reply apps/web/pages/api/rss/index.ts:54]: I wanted to, it would ideally have to come from Contentful to be translatable. Will leave it for now.

**Pull request merged**